src/components/Input.js

```diff
@@ -1,31 +1,55 @@
 import React, { Component } from 'react';
 import Tokenizer from '../parser/Tokenizer';
-import Output from './Output';
+import mermaid from 'mermaid';
 const PROGRAM = require('../ast/PROGRAM');
 export default class Input extends Component {
   constructor(props) {
     super(props);
-    this.state = { value: '', output: '' };
+    this.state = { value: '', svg: '' };
     this.handleChange = this.handleChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
   }
 
+  componentDidMount() {
+    mermaid.initialize({ startOnLoad: true });
+  }
+
   handleChange(event) {
+    let output = document.getElementById('output');
+    output.innerHTML = '';
     this.setState({ value: event.target.value });
   }
 
   handleSubmit(event) {
     event.preventDefault();
-    // Tokenize
-    Tokenizer.makeTokenizer(this.state.value);
-    console.log('Tokenizing done');
-    // Parse
-    let program = new PROGRAM();
-    program.parse();
-    console.log('Parsing done');
-    // Evaluate;
-    program.evaluate();
-    console.log('Evaluation done');
+
+    if (this.state.value) {
+      try {
+        // Tokenize
+        Tokenizer.makeTokenizer(this.state.value);
+        console.log('Tokenizing done');
```

Comment:
do we need the console logs on master?

Reply:
Mhmm I was actually thinking of that but I guess we don't really need it. I will comment them out.